File: lib/services/widget_keys.dart
// Widget data keys — Single Source of Truth for Flutter ↔ Android bridge.
//
// Every key defined here must have an identical mirror in:
//   android/.../WidgetKeys.kt
//
// The Kotlin file is a mechanical copy. If you change a key here,
// change it there too. The Kotlin file has a header comment pointing
// back to this file as the canonical source.
//
// Naming convention: all keys use `widget_` prefix + snake_case.
// New widget types add their keys in a clearly marked section below.

/// SharedPreferences file name shared between Flutter and Kotlin.
const String kWidgetPrefsName = 'HomeWidgetPreferences';

// ---------------------------------------------------------------------------
// Prayer Times
// ---------------------------------------------------------------------------
const String kKeyNextPrayerName = 'widget_next_prayer';
const String kKeyNextPrayerTime = 'widget_next_time';
const String kKeyCountdown = 'widget_countdown';

const String kKeyFajrTime = 'widget_fajr_time';
const String kKeyDhuhrTime = 'widget_dhuhr_time';
const String kKeyAsrTime = 'widget_asr_time';
const String kKeyMaghribTime = 'widget_maghrib_time';
const String kKeyIshaTime = 'widget_isha_time';
const String kKeySunriseTime = 'widget_sunrise_time';

// ---------------------------------------------------------------------------
// Quran
// ---------------------------------------------------------------------------
const String kKeyQuranSurahName = 'widget_quran_surah_name';
const String kKeyQuranSurahNumber = 'widget_quran_surah_number';
const String kKeyQuranAyah = 'widget_quran_ayah';
const String kKeyQuranPage = 'widget_quran_page';
const String kKeyQuranTotalPages = 'widget_quran_total_pages';
const String kKeyQuranProgress = 'widget_quran_progress';
const String kKeyQuranHasKhatmah = 'widget_quran_has_khatmah';

// ---------------------------------------------------------------------------
// Tasbih
// ---------------------------------------------------------------------------
const String kKeyTasbihName = 'widget_tasbih_name';
const String kKeyTasbihCount = 'widget_tasbih_count';
const String kKeyTasbihTarget = 'widget_tasbih_target';
const String kKeyTasbihId = 'widget_tasbih_id';
const String kKeyTasbihIndex = 'widget_tasbih_index';
const String kKeyTasbihTotalItems = 'widget_tasbih_total_items';

// ---------------------------------------------------------------------------
// Date / Dashboard
// ---------------------------------------------------------------------------
const String kKeyHijriDate = 'widget_hijri_date';
const String kKeyGregorianDate = 'widget_gregorian_date';
const String kKeyDayOfWeek = 'widget_day_of_week';

// ---------------------------------------------------------------------------
// Appearance
// ---------------------------------------------------------------------------
const String kKeyBgColor = 'widget_bg_color';
const String kKeyTextColor = 'widget_text_color';
const String kKeyFontSize = 'widget_font_size';

File: android/app/src/main/kotlin/com/dailyislamicwidget/daily_islamic_widget/WidgetKeys.kt
package com.dailyislamicwidget.daily_islamic_widget

/**
 * Widget data keys — Mirror of Dart-side source of truth.
 *
 * Canonical source: lib/services/widget_keys.dart
 * This file must be kept in sync with that file.
 *
 * If you add or rename a key, update BOTH files.
 */
object WidgetKeys {

    const val PREFS_NAME = "HomeWidgetPreferences"

    // Prayer Times
    const val NEXT_PRAYER_NAME = "widget_next_prayer"
    const val NEXT_PRAYER_TIME = "widget_next_time"
    const val COUNTDOWN = "widget_countdown"

    const val FAJR_TIME = "widget_fajr_time"
    const val DHUHR_TIME = "widget_dhuhr_time"
    const val ASR_TIME = "widget_asr_time"
    const val MAGHRIB_TIME = "widget_maghrib_time"
    const val ISHA_TIME = "widget_isha_time"
    const val SUNRISE_TIME = "widget_sunrise_time"

    // Quran
    const val QURAN_SURAH_NAME = "widget_quran_surah_name"
    const val QURAN_SURAH_NUMBER = "widget_quran_surah_number"
    const val QURAN_AYAH = "widget_quran_ayah"
    const val QURAN_PAGE = "widget_quran_page"
    const val QURAN_TOTAL_PAGES = "widget_quran_total_pages"
    const val QURAN_PROGRESS = "widget_quran_progress"
    const val QURAN_HAS_KHATMAH = "widget_quran_has_khatmah"

    // Tasbih
    const val TASBIH_NAME = "widget_tasbih_name"
    const val TASBIH_COUNT = "widget_tasbih_count"
    const val TASBIH_TARGET = "widget_tasbih_target"
    const val TASBIH_ID = "widget_tasbih_id"
    const val TASBIH_INDEX = "widget_tasbih_index"
    const val TASBIH_TOTAL_ITEMS = "widget_tasbih_total_items"

    // Date / Dashboard
    const val HIJRI_DATE = "widget_hijri_date"
    const val GREGORIAN_DATE = "widget_gregorian_date"
    const val DAY_OF_WEEK = "widget_day_of_week"

    // Appearance
    const val BG_COLOR = "widget_bg_color"
    const val TEXT_COLOR = "widget_text_color"
    const val FONT_SIZE = "widget_font_size"
}
